app/services/RoleService.py

```diff
@@ -0,0 +1,117 @@
+"""
+Role-related operations service.
+Provides the methods to use when interacting with a role.
+"""
+from fastapi import HTTPException
+
+from app.models.pydantic.ClassicModel import ClassicModel
+from app.models.pydantic.PermissionsModel import PydanticPermissionsModel
+from app.models.pydantic.RoleModel import PydanticRoleModel, PydanticCreateRoleModel
+from app.models.tortoise.role import RoleInDB
+from app.services.PermissionService import check_permissions
+from app.utils.enums.http_errors import CommonErrorMessages
+from app.utils.enums.permission_enums import AvailableServices, AvailableOperations
+
+
+async def get_all_roles(current_account) -> list[ClassicModel]:
+    """
+    This method retrieves all roles.
+    """
+
+    await check_permissions(AvailableServices.ACCOUNT_SERVICE,
+                            AvailableOperations.GET,
+                            current_account)
+
+    roles: list[RoleInDB] = await RoleInDB.all()
+    roles_list = []
+
+    for role in roles:
+
+        role_dict = ClassicModel(
+            name=role.name,
+            description=role.description
+        )
+
+        roles_list.append(role_dict)
+
+    return roles_list
+
+async def get_role_by_id(name: str,current_account) -> ClassicModel:
+    """
+        This method retrieves a role with given name.
+    """
+
+    await check_permissions(AvailableServices.ACCOUNT_SERVICE,
+                            AvailableOperations.GET,
+                            current_account)
+
+    role: RoleInDB | None = await RoleInDB.get_or_none(name=name)
+
+    if role is None:
+        raise HTTPException(status_code=404, detail=CommonErrorMessages.ROLE_NOT_FOUND.value)
+
+    role_dict = ClassicModel(
+        name=role.name,
+        description=role.description
+    )
+    return role_dict
+
+
+async def add_role(role: PydanticCreateRoleModel,current_account) -> None:
```

Comment:
Should be typed.
You should use a more specific DTO.

Reply:
J'utilise une DTO de création de Role, je vois pas comment je peux etre + spécifique